<?xml version="1.0"?>

<launch>
  <param name="use_sim_time" value="true"/>
  <arg name="pc_bag_file" default="/home/biped/seagate/bagfiles/2019-03-21/nclt_04_29_labeled_pc_tf.bag" />
  <!-- Visualize in rviz -->
  <!-- <node pkg="rviz" type="rviz" name="rviz"  /> -->
  <?ignore <node pkg="rosbag" type="play" name="player" output="screen" args="--clock -r 0.3 -q $(arg pc_bag_file)"/>
  ?>
  
  <!-- Painting point cloud -->
  <node pkg="segmentation_projection" type="labeled_pc_map_node" name="labeled_pc_map_node" output="screen">

      
    <remap from="~cloud_in" to="/stereo_segmentation_node/cloud_out1" />
    <remap from="~cloud_out" to="/labeled_pointcloud_pc2" />

    <param name="painting_enabled" type="bool" value="true" />
    <param name="distribution_enabled" type="bool" value="true" />
    <param name="static_frame"  type="string" value="/odom" />
    <param name="body_frame"  type="string" value="/camera_color_optical_frame" />
    <param name="stacking_visualization_enabled" type="bool" value="true" />
    <param name="path_visualization_enabled" type="bool" value="true" />
    <param name="save_pcd_enabled" type="bool" value="false" />
    <param name="color_octomap_enabled" type="bool" value="true" />

    <!-- for semantic octomap generation -->
    <param name="octomap_enabled" type="bool" value="false"/>
    <param name="octomap_num_frames" type="int" value="3000" />
    <param name="octomap_max_dist" type="int" value="10" />
    <param name="octomap_resolution" type="double" value="0.1" />
    <param name="octomap_prob_hit" type="double" value="0.53" />
    <param name="octomap_prob_miss" type="double" value="0.3" />
    <param name="octomap_occupy_thresh" type="double" value="0.52" />
  </node>

</launch>
